Add export recipes for xnnpack (#12069)

abhinaykukkadapu opened the pull request:
Summary: Enables basic export recipes for XNNPack backend described in #12069

Differential Revision: D77414795

mcr229 commented on backends/xnnpack/recipes/recipes.py:
```diff
@@ -0,0 +1,83 @@
+        name="dynamic_quant",
+        quantization_recipe=quant_recipe,
+        partitioners=[XnnpackPartitioner(config_precisions=ConfigPrecisionType.DYNAMIC_QUANT)],
+        pre_edge_transform_passes=[DuplicateDynamicQuantChainPass()],
```
can remove this

mcr229 commented on backends/xnnpack/recipes/recipes.py:
```diff
@@ -0,0 +1,83 @@
+
+
+RECIPE_MAP: dict[str, Callable[[], ExportRecipe]] = {
+    "FP32_RECIPE": get_fp32_recipe,
```
should we have a recipe for static int8? it would just be the same as dynamic config except is_dynamic = false

abhinaykukkadapu committed:
Add export recipes for xnnpack (#12069) (#12070)

Summary:
Enables basic export recipes for XNNPack backend described in https://github.com/pytorch/executorch/issues/12069

Pull Request resolved: https://github.com/pytorch/executorch/pull/12070

Test Plan:
Imported from GitHub, without a `Test Plan:` line.

Rollback Plan:

Reviewed By: metascroy, mcr229, kimishpatel

Differential Revision: D77414795

Pulled By: abhinaykukkadapu

digantdesai reviewed (commented):
Sorry a bit late, forgot to submit earlier
